drivers/overlay/ov_network.go

```diff
@@ -296,48 +295,62 @@ func (d *driver) RevokeExternalConnectivity(nid, eid string) error {
 	return nil
 }
 
-func (n *network) incEndpointCount() {
-	n.Lock()
-	defer n.Unlock()
-	n.joinCnt++
-}
+func (n *network) joinSandbox(s *subnet, restore bool, incJoinCnt bool) error {
+	networkOnce.Do(networkOnceInit)
```

Comment:
if 2 joinSanbox are happening, the first one will trigger this `networkOnceInit`, while the second one will continue forward acquiring the lock. Is it ok for the second one to go ahead also before the first `networkOnceInit` is completed?
If I look at the code I guess this will create a potential race with the `n.initSandbox(restore)` that actually if restore is `false` will try to delete the VNI
Am I missing something?

Reply:
I checked on this very early on...  once.Do() holds a mutex while doing the initialization.  If two goroutines execute the same once.Do() concurrently, the second one will block until the first one completes.